src/java/org/apache/cassandra/replication/Shard.java

```diff
@@ -158,7 +158,25 @@ private CoordinatorLog getOrCreate(CoordinatorLogId logId)
     private CoordinatorLog getOrCreate(long logId)
     {
         CoordinatorLog log = logs.get(logId);
-        return log != null
-             ? log : logs.computeIfAbsent(logId, ignore -> CoordinatorLog.create(localHostId, new CoordinatorLogId(logId), participants));
+        if (log != null)
+            return log;
+        CoordinatorLog newLog = logs.computeIfAbsent(logId, ignore -> CoordinatorLog.create(localHostId, new CoordinatorLogId(logId), participants));
```

Comment:
`CoordinatorLogId` being a POJO instead of a packing it into a long and using it that way makes me go hmm in the micro-optimization sense. It's one of those death by a thousand cuts things where no chance you can measure the impact of having it be a POJO.

Maybe we don't loop enough on maps with it as a key for it to matter and don't serialize/deserialize it that often. It might not be the right order of magnitude frequency to matter.

Reply:
Yeah, I agree, that was changed from long to CoordinatorLogId alongside the recent SRP changes.